aws-api/src/main/java/com/amplifyframework/api/aws/DomainType.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.api.aws;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * The type of domain specified in API endpoint.
+ */
+public enum DomainType {
+    /**
+     * Invalid or unsupported domain type.
+     */
+    INVALID,
+
+    /**
+     * Standard domain type composed of randomly generated 26 char subdomain.
+     */
+    STANDARD,
+
+    /**
+     * Custom domain defined by user. It must be "https" and have path "graphql" in URL.
+     */
+    CUSTOM;
+
+    private static final String STANDARD_ENDPOINT_REGEX =
```

Comment:
is the endpoint format spec documented somewhere?

Reply:
Endpoint and protocol for appsync service is outlined here:
https://docs.aws.amazon.com/general/latest/gr/appsync.html
The regex of generating 26 alphanumeric character unique id portion is what is used by appsync team and used here as is.

Reply:
Can we have that link directly commented in code for easy reference?